here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/models/naksha/EventHandler.java

```diff
@@ -123,9 +123,15 @@ public void setExtensionId(@Nullable String extensionId) {
   @AvailableSince(NakshaVersion.v2_0_7)
   public @NotNull IEventHandler newInstance(@NotNull INaksha naksha, @NotNull EventTarget<?> eventTarget) {
     //noinspection unchecked
-    final Fe3<IEventHandler, INaksha, EventHandler, EventTarget<?>> constructor =
-        (Fe3<IEventHandler, INaksha, EventHandler, EventTarget<?>>)
-            getEventHandlerConstructor(getClassName(), EventHandler.class, eventTarget.getClass());
+    final Fe3<IEventHandler, INaksha, EventHandler, EventTarget<?>> constructor;
+    if (this.extensionId == null || this.extensionId.isEmpty() || "null".equalsIgnoreCase(this.extensionId)) {
+      constructor = (Fe3<IEventHandler, INaksha, EventHandler, EventTarget<?>>)
+          getEventHandlerConstructor(getClassName(), EventHandler.class, eventTarget.getClass());
+    } else {
+      ClassLoader extClassLoader = naksha.getClassLoader(this.extensionId);
+      constructor = (Fe3<IEventHandler, INaksha, EventHandler, EventTarget<?>>) getEventHandlerConstructor(
+          getClassName(), EventHandler.class, eventTarget.getClass(), extensionId, extClassLoader);
+    }
     try {
       return constructor.call(naksha, this, eventTarget);
     } catch (Exception e) {
```

Comment:
This code is now subject to multiple possible exceptions. Put entire function into try-catch block.
When exception occurs, let's log the error as well.
`log.error("Exception loading constructor for EventHandler id {}, extensionId {}.", getId(), this.extensionId, e);`

Reply:
fixed